### DISCUSSION TITLE
How to iterate through pluginOptions?

### DISCUSSION THREAD

**Discussion by segfault-bilibili** [shadowsocks/shadowsocks-android#2858] [Q&A] [answered]
I'm not sure whether something like this will work:
```Java
Set<String> keys = pluginOptions.getKeys();
for (String key : keys) {
    String value = pluginOptions.get(key);
// do something
}
```

**Comment by Mygod** [comment, marked as answer]
The same way that you iterate over any `HashMap`.

**Reply by segfault-bilibili** [reply]
Is it `HashMap<String, String>` or something else?

**Reply by segfault-bilibili** [reply]
Hmm, actually I no longer iterate through it... Now I just use the `toString()` method and `pluginOptions(String s)` constructor.
